ckanext/notify/actions.py

```diff
@@ -0,0 +1,474 @@
+import ckan.logic as logic
+import ckan.model as model
+import ckan.lib.base as base
+import ckan.plugins as plugins
+import ckan.lib.mailer as mailer
+import constants
+import validator
+import json
+import requests
+import db
+
+
+from ckan.common import _, config
+from socket import error as socket_error
+
+toolkit = plugins.toolkit
+c = toolkit.c
+
+ValidationError = logic.ValidationError
+
+
+def _dictize_slack_details(slack_details):
+
+    # Convert the slack details into a dict
+    data_dict = {
+        'id': slack_details.id,
+        'webhook_url': slack_details.webhook_url,
+        'slack_channel': slack_details.slack_channel,
```

Comment:
I'm unsure but why do we need the slack_channel detail? Isn't `webhook_url` enough?

Reply:
@andela-iukwuoma ☝️

Reply:
Actually, `webhook_url` is enough. We felt there was a need to display the `slack_channel` for the sake of the user who might not remember the `slack_channel` linked to the `webhook_url`.

Reply:
@DavidLemayian

Reply:
@andela-iukwuoma Hmm.. As a "label"?

I'm wondering if the cost of this placebo outweighs the frustration of a user not knowing what format or whether it's the channel or account name etc. It's the first thing that I though about when I saw it.

I'd say we keep it for now and review user feedback / use.